test/widgets/message_list_test.dart

```diff
@@ -909,6 +921,35 @@ void main() {
           expectedTitle: zulipLocalizations.errorMarkAsReadFailedTitle,
           expectedMessage: 'Oops');
       });
+
+      testWidgets('MessageList does not shift down when mark as read button is pressed', (WidgetTester tester) async {
+        final message = eg.streamMessage(flags: []);
+        final unreadMsgs = eg.unreadMsgs(streams:[
+          UnreadStreamSnapshot(topic: message.subject, streamId: message.streamId, unreadMessageIds: [message.id])
+        ]);
+        await setupMessageListPage(tester, messages: [message], unreadMsgs: unreadMsgs);
+        check(isMarkAsReadButtonVisible(tester)).isTrue();
+
+        // Get the top position of the first message item before the mark as read button is pressed
+        // and then after the button is pressed. The top position should remain the same.
+        // This is to ensure that the message list does not shift down when the mark as read button is pressed.
+        final firstMessageItemTopPositionBefore = tester.getTopLeft(find.byType(MessageItem).first).dy;
+
+        store.handleEvent(UpdateMessageFlagsAddEvent(
+          id: 1,
+          flag: MessageFlag.read,
+          messages: [message.id],
+          all: false,
+        ));
+
+        await tester.pumpAndSettle();
+        check(isMarkAsReadButtonVisible(tester)).isFalse();
+
+        final firstMessageItemTopPositionAfter = tester.getTopLeft(find.byType(MessageItem).first).dy;
+
+        check(firstMessageItemTopPositionBefore).equals(firstMessageItemTopPositionAfter);
```

Comment:
I tried to use the` .single `method, but that returns an `Element` rather than a `Finder` and the `tester.getTopLeft` method takes a `Finder`. So in order to explicitly show that there is only one MessageItem on this test, I have added a check that checks if the number of MessageItem widgets found is equal to 1. If there is a better way to handle this test I am open to suggestions.

Reply:
Hmm, I see. Sure, this works.